src/lib.rs

```diff
@@ -146,6 +158,7 @@ pub struct Printer<'a, Writer: Write> {
     buffer_line: Vec<u8>,
     writer: &'a mut Writer,
     show_color: bool,
+    use_8_bit_color: bool,
```

Comment:
I would prefer if we use an enum instead of a bool. Maybe
```suggestion
enum ColorMode {
  Mode16,
  Mode8Bit,
}
```

Maybe it also makes sense to combine it with the `show_color` bool?